client/driver/exec.go

```diff
@@ -41,10 +47,40 @@ func (d *ExecDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool,
 }
 
 func (d *ExecDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle, error) {
-	// Get the command
+	// Get the command to be ran, or if omitted, download an artifact for
+	// execution. Currently a supplied command takes precedence, and an artifact
+	// is only downloaded if no command is supplied
 	command, ok := task.Config["command"]
 	if !ok || command == "" {
-		return nil, fmt.Errorf("missing command for exec driver")
+		source, sok := task.Config["artifact_source"]
```

Comment:
My understanding was we were targeting isolated binaries, similar to the Java jars or the Qemu images. 

Downloading and extracting zip/tars wouldn't require too much more, but we may want to think more on that. We would need to expand it further to both support an artifact that's a package (zip/tar) and a command that invokes an arbitrary _thing_ inside of it. Not hard necessarily, but worth discussion if that's what we want to support in this first pass